db/models.py

```diff
@@ -42,11 +54,92 @@ def __str__(self) -> str:
 class MovieSession(models.Model):
     show_time = models.DateTimeField()
     cinema_hall = models.ForeignKey(
-        to=CinemaHall, on_delete=models.CASCADE, related_name="movie_sessions"
+        to=CinemaHall,
+        on_delete=models.CASCADE,
+        related_name="movie_sessions"
     )
     movie = models.ForeignKey(
-        to=Movie, on_delete=models.CASCADE, related_name="movie_sessions"
+        to=Movie,
+        on_delete=models.CASCADE,
+        related_name="movie_sessions"
     )
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(
+        AUTH_USER_MODEL,
+        on_delete=models.CASCADE,
+        related_name="orders"
+    )
+
+    class Meta:
+        ordering = ["-created_at"]
+
+    def __str__(self) -> str:
+        return str(self.created_at)
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey(
+        "MovieSession",
+        on_delete=models.CASCADE,
+        related_name="tickets"
+    )
+    order = models.ForeignKey(
+        "Order",
+        on_delete=models.CASCADE,
+        related_name="tickets"
+    )
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    def __str__(self) -> str:
+        return (
+            f"{self.movie_session.movie.title} "
+            f"{self.movie_session.show_time} "
+            f"(row: {str(self.row)}, seat: {str(self.seat)})"
+        )
+
+    def clean(self) -> None:
+        max_rows = self.movie_session.cinema_hall.rows
+        if self.row < 1 or self.row > max_rows:
+            raise ValidationError({
+                "row": [
+                    f"row number must be in available range: "
+                    f"(1, rows): (1, {max_rows})"
+                ]
+            })
+
+        max_seats = self.movie_session.cinema_hall.seats_in_row
+        if self.seat < 1 or self.seat > max_seats:
+            raise ValidationError({
+                "seat": [
+                    f"seat number must be in available range: "
+                    f"(1, seats_in_row): (1, {max_seats})"
+                ]
+            })
+
+    class Meta:
+        constraints = [
+            UniqueConstraint(
+                name="unique_ticket", fields=["row", "seat", "movie_session"]
+            )
+        ]
+
+    def save(
+            self,
+            force_insert: bool = False,
+            force_update: bool = False,
+            using: Any = None,
+            update_fields: Any = None
+    ) -> None:
+        self.full_clean()
+        return super(Ticket, self).save()
```

Comment:
do we need to provide arrangements to` super()` ?, and we definitely need to provide arguments to `save()`